test/e2e/singlecluster/kuberay_test.go

```diff
@@ -204,6 +301,72 @@ var _ = ginkgo.Describe("Kuberay", func() {
 			}, util.VeryLongTimeout, util.Interval).Should(gomega.Succeed())
 		})
 
+		ginkgo.By("Waiting for 3 pods in rayjob namespace", func() {
+			// 3 rayjob pods: head, worker, submitter job
+			gomega.Eventually(func(g gomega.Gomega) {
+				podList := &corev1.PodList{}
+				g.Expect(k8sClient.List(ctx, podList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				g.Expect(podList.Items).To(gomega.HaveLen(3), "Expected exactly 3 pods in rayjob namespace")
+				// Get worker pod names and check count
+				workerPodNames := getRunningWorkerPodNames(podList)
+				g.Expect(workerPodNames).To(gomega.HaveLen(1), "Expected exactly 1 pod with 'workers' in the name")
+
+				// Store initial pod names for later verification
+				initialPodNames = workerPodNames
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("Waiting for 2 workloads", func() {
+			// 2 workloads: one for the ray cluster, another for the submitter job created by the ray job
+			gomega.Eventually(func(g gomega.Gomega) {
+				workloadList := &kueue.WorkloadList{}
+				g.Expect(k8sClient.List(ctx, workloadList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				g.Expect(workloadList.Items).To(gomega.HaveLen(2), "Expected exactly 2 workloads")
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("Waiting for 5 workers due to scaling up", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				podList := &corev1.PodList{}
+				g.Expect(k8sClient.List(ctx, podList, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				// Get worker pod names and check count
+				currentPodNames := getRunningWorkerPodNames(podList)
+				g.Expect(currentPodNames).To(gomega.HaveLen(5), "Expected exactly 5 pods with 'workers' in the name")
+
+				// Verify that the current pod names are a superset of the initial pod names
+				g.Expect(verifyPodNamesAreSuperset(currentPodNames, initialPodNames)).To(gomega.BeTrue(),
+					"Current worker pod names should be a superset of initial pod names. "+
+						"Initial pods: %v, Current pods: %v", initialPodNames, currentPodNames)
```

Comment:
Nit, instead of the helper functions let's use the gomega way:

```
g.Expect(currentPodNames.UnsortedList()).To(gomega.ContainElements(initialPodNames.UnsortedList()))
```
This is less code, and better failure message when it happens